backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -0,0 +1,205 @@
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      session.beginTransaction();
+      final String rootSha = setBlobs.apply(session);
+      long timeCreated = new Date().getTime();
+      if (App.getInstance().getStoreClientCreationTimestamp() && commit.getDateCreated() != 0L) {
+        timeCreated = commit.getDateCreated();
+      }
+      final String commitSha = generateCommitSHA(rootSha, commit, timeCreated);
+      Commit internalCommit =
+          Commit.newBuilder()
+              .setDateCreated(timeCreated)
+              .setAuthor(author)
+              .setMessage(commit.getMessage())
+              .setCommitSha(commitSha)
+              .build();
+      CommitEntity commitEntity =
+          new CommitEntity(
+              getRepository.apply(session),
+              getCommits(session, commit.getParentShasList()),
+              internalCommit,
+              rootSha);
+      session.saveOrUpdate(commitEntity);
+      session.getTransaction().commit();
+      return Response.newBuilder().setCommit(commitEntity.toCommitProto()).build();
+    }
+  }
+
+  @Override
+  public ListCommitsRequest.Response listCommits(
+      ListCommitsRequest request, RepositoryFunction getRepository) throws ModelDBException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      RepositoryEntity repository = getRepository.apply(session);
+
+      StringBuilder commitQueryBuilder =
+          new StringBuilder(
+              "SELECT cm FROM "
+                  + CommitEntity.class.getSimpleName()
+                  + " cm LEFT JOIN cm.repository repo WHERE repo.id = :repoId ");
+      if (!request.getCommitBase().isEmpty()) {
+        CommitEntity baseCommitEntity =
+            Optional.ofNullable(session.get(CommitEntity.class, request.getCommitBase()))
+                .orElseThrow(
+                    () ->
+                        new ModelDBException(
+                            "Couldn't find base commit by sha : " + request.getCommitBase(),
+                            Code.NOT_FOUND));
+        Long baseTime = baseCommitEntity.getDate_created();
+        commitQueryBuilder.append(" AND cm.date_created >= " + baseTime);
+      }
+
+      if (!request.getCommitHead().isEmpty()) {
+        CommitEntity headCommitEntity =
+            Optional.ofNullable(session.get(CommitEntity.class, request.getCommitHead()))
+                .orElseThrow(
+                    () ->
+                        new ModelDBException(
+                            "Couldn't find head commit by sha : " + request.getCommitHead(),
+                            Code.NOT_FOUND));
+        Long headTime = headCommitEntity.getDate_created();
+        commitQueryBuilder.append(" AND cm.date_created <= " + headTime);
+      }
+
+      Query<CommitEntity> commitEntityQuery =
+          session.createQuery(
+              commitQueryBuilder.append(" ORDER BY cm.date_created ASC").toString());
```

Comment:
```suggestion
              commitQueryBuilder.append(" ORDER BY cm.date_created DESC").toString());
```
It should be in reverse order to get the last created commit by:
commits?pagination.page_number=1&pagination.page_limit=1
Please confirm

Reply:
@ravishetye @ad-47